# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/clients/manager/ClientManager.cpp
# Commit: fb720435215f
# Full Hash: fb720435215fa5be3b2bd2101f3a54382954b70c
# Author: Andrew Sutherland <asutherland@asutherland.org>
# Date: 2024-10-24 09:44:34
# Regressor Bug: 1544232
# File Overlap Count: 4
# Description:
#   Bug 1544232 - Limit lifetime extension of SWs by SWs to the sender's lifetime. r=edenchuang
#   
#   This patch introduces an explicit concept of lifetimes with mechanisms
#   in place so that actions taken by Clients (windows or non-ServiceWorker
#   orkers) will extend the lifetime of a ServiceWorker, but a ServiceWorker
# ==============================================================================

diff -r 25452f6623a0 -r fb720435215f dom/clients/manager/ClientManager.cpp
--- a/dom/clients/manager/ClientManager.cpp	Thu Oct 24 03:02:41 2024 +0000
+++ b/dom/clients/manager/ClientManager.cpp	Thu Oct 24 03:02:42 2024 +0000
@@ -112,13 +112,17 @@
     // value.  Instead return a shutdown ClientSource with a zero'd id.
     // This should be exceptionally rare, if it happens at all.
     id.Clear();
-    ClientSourceConstructorArgs args(id, aType, aPrincipal, TimeStamp::Now());
+    ClientSourceConstructorArgs args(id, Nothing(), aType, aPrincipal,
+                                     TimeStamp::Now(), VoidCString(),
+                                     FrameType::None);
     UniquePtr<ClientSource> source(new ClientSource(this, aEventTarget, args));
     source->Shutdown();
     return source;
   }
 
-  ClientSourceConstructorArgs args(id, aType, aPrincipal, TimeStamp::Now());
+  ClientSourceConstructorArgs args(id, Nothing(), aType, aPrincipal,
+                                   TimeStamp::Now(), VoidCString(),
+                                   FrameType::None);
   UniquePtr<ClientSource> source(new ClientSource(this, aEventTarget, args));
 
   if (IsShutdown()) {
@@ -135,9 +139,10 @@
     const ClientInfo& aClientInfo, nsISerialEventTarget* aEventTarget) {
   NS_ASSERT_OWNINGTHREAD(ClientManager);
 
-  ClientSourceConstructorArgs args(aClientInfo.Id(), aClientInfo.Type(),
-                                   aClientInfo.PrincipalInfo(),
-                                   aClientInfo.CreationTime());
+  ClientSourceConstructorArgs args(
+      aClientInfo.Id(), aClientInfo.AgentClusterId(), aClientInfo.Type(),
+      aClientInfo.PrincipalInfo(), aClientInfo.CreationTime(),
+      aClientInfo.URL(), aClientInfo.FrameType());
   UniquePtr<ClientSource> source(new ClientSource(this, aEventTarget, args));
 
   if (IsShutdown()) {
@@ -344,7 +349,8 @@
     return Nothing();
   }
 
-  return Some(ClientInfo(id, aType, principalInfo, TimeStamp::Now()));
+  return Some(ClientInfo(id, Nothing(), aType, principalInfo, TimeStamp::Now(),
+                         ""_ns, FrameType::None));
 }
 
 // static